### DISCUSSION TITLE
How does rocm-arch deal with library paths (ld.so.conf, LD_LIBRARY_PATH)

### DISCUSSION THREAD

**Discussion by sofiageo** [rocm-arch/rocm-arch#675] [Support] [answered]
I will admit I haven't installed rocm-arch on my PC yet. The reason is mostly that I'm maintaining a binary package for ROCm based on the Ubuntu release, and it would take a lot of effort and resources to uninstall / install all the time.

I'm trying to understand what is the correct way to provide the library paths to software like Pytorch / Tensorflow / JuliaGPU etc. I thought adding the paths to `ld.so.conf` would be enough but it doesn't seem to be working. Only when I set the `LD_LIBRARY_PATH` these applications start to work properly. So my question is what I'm doing wrong and how does rocm-arch deals with it?

**Comment by sofiageo** [comment]
My assumption after a lot of investigation and trial and error is that the RUNPATH of the executables messes with this, because the search priority for runpath is higher in the [linker order](https://man7.org/linux/man-pages/man8/ld.so.8.html). So `ld.so.conf` should work fine as long as the executables are not pointing to a wrong library path.

**Comment by acxz** [comment, marked as answer]
> I thought adding the paths to ld.so.conf would be enough but it doesn't seem to be working.

This is how we do it in the PKGBUILDs.
